x/liquidity/spec/04_messages.md

```diff
@@ -1,95 +1,101 @@
+Withdraw coins in batch from liquidity pool with the `MsgWithdrawWithinBatch` message.
 
 ```go
 type MsgWithdrawWithinBatch struct {
-	WithdrawerAddress string         // account address of the origin of this message
-	PoolId            uint64         // id of the liquidity pool where this message is belong to
-	PoolCoin          sdk.Coin       // pool coin sent for reserve coin withdraw
+    WithdrawerAddress string         // account address of the origin of this message
+    PoolId            uint64         // id of the liquidity pool to withdraw the coins from
+    PoolCoin          sdk.Coin       // pool coin sent for reserve coin withdrawal
 }
 ```
 
-**Validity check**
+## Validity checks
+
+The MsgWithdrawWithinBatch message performs validity checks. The transaction that is triggered with the `MsgWithdrawWithinBatch` message fails if:
 
-- `MsgWithdrawWithinBatch` failes if
-  - `Withdrawer` address does not exist
-  - `PoolId` does not exist
-  - if the denom of `PoolCoin` are not equal to the `PoolCoinDenom` of the `LiquidityPool` with given `PoolId`
-  - if the balance of `Depositor` does not have enough amount of coins for `PoolCoin`
+- `Withdrawer` address does not exist
+- `PoolId` does not exist
+- The denom of `PoolCoin` are not equal to the `PoolCoinDenom` of the `LiquidityPool`
+- The balance of `Depositor` does not have enough coins for `PoolCoin`
 
 ## MsgSwapWithinBatch
 
-// Swap offer coin with demand coin from the specified liquidity pool with the given order price.
+Swap coins between liquidity pools in batch with the `MsgSwapWithinBatch` message.
+
+Offer coins are swapped with demand coins for the given order price.
 
 ```go
 type MsgSwapWithinBatch struct {
-	SwapRequesterAddress string     // account address of the origin of this message
-	PoolId               uint64     // id of the liquidity pool where this message is belong to
-	SwapTypeId           uint32     // swap type id of this swap message, default 1: InstantSwap, requesting instant swap
-	OfferCoin            sdk.Coin   // offer coin of this swap message
-	DemandCoinDenom      string     // denom of demand coin of this swap message
-	OfferCoinFee         sdk.Coin   // offer coin fee for pay fees in half offer coin
-	OrderPrice           sdk.Dec    // limit order price for the order, the price is the exchange ratio of X/Y where X is the amount of the first coin and Y is the amount of the second coin when their denoms are sorted alphabetically
+    SwapRequesterAddress string     // account address of the origin of this message
+    PoolId               uint64     // id of the liquidity pool
+    SwapTypeId           uint32     // swap type id of this swap message, default 1: InstantSwap, requesting instant swap
+    OfferCoin            sdk.Coin   // offer coin of this swap
+    DemandCoinDenom      string     // denom of demand coin of this swap
+    OfferCoinFee         sdk.Coin   // offer coin fee for pay fees in half offer coin
```

Comment:
what is "half offer coin" 
can we explain please?

Reply:
You could check the context here https://github.com/tendermint/liquidity/issues/41